File: swift/Sources/FluentCards/InputTimeBuilder.swift
/// InputTimeBuilder builds an Input.Time Adaptive Card element.
public final class InputTimeBuilder {
    var data: [String: Any] = ["type": "Input.Time", "id": ""]

    public init() {}

    @discardableResult
    public func withID(_ id: String) -> InputTimeBuilder {
        data["id"] = id
        return self
    }

    @discardableResult
    public func withLabel(_ label: String) -> InputTimeBuilder {
        data["label"] = label
        return self
    }

    @discardableResult
    public func withPlaceholder(_ placeholder: String) -> InputTimeBuilder {
        data["placeholder"] = placeholder
        return self
    }

    @discardableResult
    public func withValue(_ value: String) -> InputTimeBuilder {
        data["value"] = value
        return self
    }

    /// Sets the minimum time (format: HH:MM).
    @discardableResult
    public func withMin(_ min: String) -> InputTimeBuilder {
        data["min"] = min
        return self
    }

    /// Sets the maximum time (format: HH:MM).
    @discardableResult
    public func withMax(_ max: String) -> InputTimeBuilder {
        data["max"] = max
        return self
    }

    @discardableResult
    public func withIsRequired(_ isRequired: Bool) -> InputTimeBuilder {
        data["isRequired"] = isRequired
        return self
    }

    @discardableResult
    public func withErrorMessage(_ errorMessage: String) -> InputTimeBuilder {
        data["errorMessage"] = errorMessage
        return self
    }

    @discardableResult
    public func withSpacing(_ spacing: Spacing) -> InputTimeBuilder {
        data["spacing"] = spacing.rawValue
        return self
    }

    public func build() -> Card {
        return data
    }
}

File: java/src/main/java/io/fluentcards/InputTimeBuilder.java
package io.fluentcards;

import java.util.*;

public class InputTimeBuilder {
    private final LinkedHashMap<String, Object> data;

    public InputTimeBuilder() {
        data = new LinkedHashMap<>();
        data.put("type", "Input.Time");
        data.put("id", "");
    }

    public InputTimeBuilder withId(String id) {
        data.put("id", id);
        return this;
    }

    public InputTimeBuilder withLabel(String label) {
        data.put("label", label);
        return this;
    }

    public InputTimeBuilder withPlaceholder(String placeholder) {
        data.put("placeholder", placeholder);
        return this;
    }

    public InputTimeBuilder withValue(String value) {
        data.put("value", value);
        return this;
    }

    public InputTimeBuilder withMin(String min) {
        data.put("min", min);
        return this;
    }

    public InputTimeBuilder withMax(String max) {
        data.put("max", max);
        return this;
    }

    public InputTimeBuilder withIsRequired(boolean isRequired) {
        data.put("isRequired", isRequired);
        return this;
    }

    public InputTimeBuilder withErrorMessage(String errorMessage) {
        data.put("errorMessage", errorMessage);
        return this;
    }

    public InputTimeBuilder withSpacing(Spacing spacing) {
        data.put("spacing", spacing.getValue());
        return this;
    }

    public Map<String, Object> build() {
        return data;
    }
}
